packages/core/src/components/button/button.stories.tsx

```diff
@@ -128,6 +128,14 @@ export default {
         defaultValue: { summary: 'none' },
       },
     },
+
+    tdsAriaLabel: {
+      name: 'Aria Label',
+      description: 'Value to be used for the aria-label attribute',
+      control: {
+        type: 'text',
+      },
+    },
```

Comment:
@max-umain you can make this Storybook control conditional, aka it should appear only if onlyIcon is set to true.

Take a look on examples in this story file where you find:
`if: { arg: 'size', neq: 'Extra small' },`

and 
`if: { arg: 'onlyIcon', truthy: false },`

Examples are here just to show you what to look for and try out.